airflow/www/views.py

```diff
@@ -1421,6 +1421,7 @@ def duration(self):
 
         tis = dag.get_task_instances(
             session, start_date=min_date, end_date=base_date)
+        tis.sort(key=lambda x: x.execution_date)
```

Comment:
Why not change dag.get_task_instances to return it sorted on execution date? That saves memory and is faster as it will be handled by the database